pkg/importer/dataStream.go

```diff
@@ -467,8 +483,18 @@ func closeReaders(readers []Reader) (rtnerr error) {
 	return rtnerr
 }
 
+func (d dataStream) isHTTPQcow2() bool {
+	// assuming len(d.Readers) == 3 is [http, mr, mr]
+	// should prob get rid of the redundant MultiReader
```

Comment:
agree. Where is the extra mr coming from?

Reply:
the loop in constructReaders() always adds a MultiReader to the stack, when matchHeader() returns a match.  This is a problem in the qcow2 case because unlike the other possible matches, no qcow2 reader is added to the stack.

I think there is a pretty straightforward fix for this.

Reply:
Ok, my bad and thanks for finding this. If you want to fix this in a separate pr let me know asap, else I'll fix it. Thanks.

Reply:
@jeffvance I spent a bit of time looking into this earlier.  Turns out a fix is not as simple as I thought.  I think a separate PR would be appropriate.

Reply:
I just added pr #339. Thanks for mentioning this as it turned out to be a bug in `constructReaders` and the unit test too.